lib/datasource/golang-version/index.ts

```diff
@@ -0,0 +1,107 @@
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import { cache } from '../../util/cache/package/decorator';
+import { regEx } from '../../util/regex';
+import { isVersion, id as semverVersioningId } from '../../versioning/semver';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+const lineTerminationRegex = regEx(`\r?\n`);
+const releaseBeginningChar = '{';
+const releaseTerminationChar = '},';
+const releaseDateRegex = regEx(
+  `Date\\{(?<year>\\d+),\\s+(?<month>\\d+),\\s+(?<day>\\d+)\\}`
+);
+const releaseVersionRegex = regEx(
+  `Version\\{(?<versionMajor>\\d+),\\s+(?<versionMinor>\\d+),\\s+(?<patch>\\d+)\\}`
+);
+
+export class GolangVersionDatasource extends Datasource {
+  static readonly id = 'golang-version';
+
+  constructor() {
+    super(GolangVersionDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = [
+    'https://raw.githubusercontent.com/golang/website/',
+  ];
+
+  override readonly customRegistrySupport = false;
+
+  override readonly defaultVersioning = semverVersioningId;
+
+  @cache({ namespace: `datasource-${GolangVersionDatasource.id}`, key: 'all' })
+  async getReleases({
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const res: ReleaseResult = {
+      homepage: 'https://go.dev/',
+      sourceUrl:
+        'https://github.com/golang/website/blob/master/internal/history/release.go',
```

Comment:
Can we reference the go source repo instead? Or a go repo with changelogs available?